assets/src/components/mapMarkers.tsx

```diff
@@ -0,0 +1,311 @@
+    trainVehicle.latitude,
+    trainVehicle.longitude,
+  ]
+  const icon: Leaflet.DivIcon = makeTrainVehicleIcon(trainVehicle)
+  return <Marker position={position} icon={icon} />
+}
+
+export const strokeOptions = ({ color }: Shape): object =>
+  color
+    ? {
+        color,
+        opacity: 1.0,
+        weight: 4,
+      }
+    : {
+        color: "#4db6ac",
+        opacity: 0.6,
+        weight: 6,
+      }
+
+const StopMarker = React.memo(({ stop }: { stop: Stop }) => (
+  <CircleMarker
+    className="m-vehicle-map__stop"
+    center={[stop.lat, stop.lon]}
+    radius={3}
+  >
+    <Popup className="m-vehicle-map__stop-tooltip">
+      {stop.name}
+      {inTestGroup(MAP_BETA_GROUP_NAME) && (
+        <StreetViewButton
+          latitude={stop.lat}
+          longitude={stop.lon}
+        ></StreetViewButton>
+      )}
+    </Popup>
+  </CircleMarker>
+))
+
+export const StationMarker = React.memo(
+  ({ station, iconSize }: { station: Stop; iconSize: StationIconSize }) => {
+    const iconSizeLength = iconSize === StationIconSize.small ? 12 : 16
+
+    return (
+      <Marker
+        position={[station.lat, station.lon]}
+        icon={stationLeafletIcon({ size: iconSizeLength })}
+      >
+        <Tooltip
+          className="m-vehicle-map__station-tooltip"
+          direction={"top"}
+          offset={[0, -(iconSizeLength / 2 + 8)]}
+        >
+          {station.name}
+        </Tooltip>
+      </Marker>
+    )
+  }
+)
+
+export enum StationIconSize {
+  small,
+  large,
+}
+
+export const RouteStopMarkers = ({
+  stops,
+  iconSize,
+}: {
+  stops: Stop[]
+  iconSize: StationIconSize
+}): JSX.Element => {
+  const seenStopIds = new Set<StopId>()
+  // Keep the first occurance of each stop when there are duplicates
+  const uniqueStops: Stop[] = stops.flatMap((stop) => {
+    if (!seenStopIds.has(stop.id)) {
+      seenStopIds.add(stop.id)
+      return [stop]
+    }
+    return []
+  })
+
+  return (
+    <>
+      {uniqueStops.map((stop) =>
+        stop.locationType === LocationType.Station ? (
+          <StationMarker station={stop} iconSize={iconSize} key={stop.id} />
+        ) : (
+          <StopMarker stop={stop} key={stop.id} />
+        )
+      )}
+    </>
+  )
+}
+
+export const RouteShape = React.memo(({ shape }: { shape: Shape }) => {
+  const positions: LatLngExpression[] = shape.points.map((point) => [
+    point.lat,
+    point.lon,
+  ])
+  return (
+    <Polyline
+      className="m-vehicle-map__route-shape"
+      positions={positions}
+      {...strokeOptions(shape)}
+    />
+  )
+})
+
+const garageLeafletIcon = Leaflet.divIcon({
+  html: garageIcon,
+  className: "m-garage-icon",
+  iconAnchor: new Leaflet.Point(10, 25),
+})
+
+const stationLeafletIcon = ({ size }: { size: number }): Leaflet.DivIcon => {
+  return Leaflet.divIcon({
+    html: stationIcon,
+    className: "m-station-icon",
+    iconSize: [size, size],
+    iconAnchor: [size / 2, size / 2],
+  })
+}
```

Comment:
**nitpick (non-blocking):** it might be better to order this with the other stop / station icon stuff rather than in between garage-related components.